python/foglamp/plugins/south/cc2650async/cc2650async.py

```diff
@@ -0,0 +1,379 @@
+
+        data['notification_handles'] = tag.get_notification_handles()
+        data['characteristics'] = sensortag_characteristics
+        data['tag'] = tag
+        _LOGGER.info('SensorTagCC2650 {} Async fetching initialized'.format(bluetooth_adr))
+
+    return data
+
+
+def plugin_start(handle):
+    """ Extracts data from the sensor and returns it in a JSON document as a Python dict.
+
+    Available for async mode only.
+
+    Args:
+        handle: handle returned by the plugin initialisation call
+    Returns:
+        returns a sensor reading in a JSON document, as a Python dict, if it is available
+        None - If no reading is available
+    Raises:
+        TimeoutError
+    """
+    async def save_data():
+        if 'tag' not in handle:
+            return
+        time_stamp = str(datetime.datetime.now(tz=datetime.timezone.utc))
+        data = {
+            'asset': 'TI Sensortag CC2650',
+            'timestamp': time_stamp,
+            'key': str(uuid.uuid4()),
+            'readings': {}
+        }
+        bluetooth_adr = handle['bluetoothAddress']['value']
+        tag = handle['tag']
+        object_temp_celsius = None
+        ambient_temp_celsius = None
+        lux_luminance = None
+        rel_humidity = None
+        rel_temperature = None
+        bar_pressure = None
+        movement = None
+        battery_level = None
+        keypress_state = None
+
+        try:
+            if not tag.is_connected:
+                raise RuntimeError
+
+            # Enable notification
+            for notification_handle in handle['notification_handles']:
+                tag.char_write_cmd(notification_handle, notification_enable)
+
+            # Enable sensors
+            tag.char_write_cmd(handle['characteristics']['temperature']['configuration']['handle'], char_enable)
+            tag.char_write_cmd(handle['characteristics']['luminance']['configuration']['handle'], char_enable)
+            tag.char_write_cmd(handle['characteristics']['humidity']['configuration']['handle'], char_enable)
+            tag.char_write_cmd(handle['characteristics']['pressure']['configuration']['handle'], char_enable)
+            tag.char_write_cmd(handle['characteristics']['movement']['configuration']['handle'], movement_enable)
+
+            # TODO: How to implement CTRL-C or terminate process?
+            debug_cnt = 0  # Used only for debugging. debug_cnt should be set to 0 in production.
+            cnt = 0
+            while True:
+                time_stamp = str(datetime.datetime.now(tz=datetime.timezone.utc))
+                try:
+                    pattern_index = tag.con.expect('Notification handle = .*? \r', timeout=4)
+                except pexpect.TIMEOUT:
+                    _LOGGER.error("SensorTagCC2650 {} async timeout")
+                    print("TIMEOUT exception!")
+                    break
+
+                # If succesfull, then pattern_index should appear at col 0
+                if pattern_index == 0:
```

Comment:
It would be useful to add an explanation as to the format you are processing here.